### ISSUE TITLE
Jupyter C++ kernel dies repeatedly 

### ISSUE THREAD

**Issue by Yancey2126** [vgvassilev/cling#171]
I installed cling and libgcc following this article:
http://shuvomoy.github.io/blog/programming/2016/08/04/Cpp-kernel-for-Jupyter.html

However, after running jupyter notebook, I can see C++ except that the kernel can never start and died repeatedly.

`KernelRestarter: restarting kernel (4/5)
kernel 8d8df6db-061d-4d07-87f4-b230de31a007 restarted
Traceback (most recent call last):
  File "/usr/local/bin/jupyter-cling-kernel", line 6, in <module>
    exec(compile(open(__file__).read(), __file__, 'exec'))
  File "/Users/yangchen/Desktop/C++/clingMac/share/cling/Jupyter/kernel/scripts/jupyter-cling-kernel", line 4, in <module>
    main()
  File "/Users/yangchen/Desktop/C++/clingMac/share/cling/Jupyter/kernel/clingkernel.py", line 354, in main
    ClingKernelApp.launch_instance()
  File "/Library/Python/2.7/site-packages/traitlets/config/application.py", line 657, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-121>", line 2, in initialize
  File "/Library/Python/2.7/site-packages/traitlets/config/application.py", line 87, in catch_config_error
    return method(app, *args, **kwargs)
  File "/Library/Python/2.7/site-packages/ipykernel/kernelapp.py", line 457, in initialize
    self.init_kernel()
  File "/Library/Python/2.7/site-packages/ipykernel/kernelapp.py", line 368, in init_kernel
    user_ns=self.user_ns,
  File "/Library/Python/2.7/site-packages/traitlets/config/configurable.py", line 412, in instance
    inst = cls(*args, **kwargs)
  File "/Users/yangchen/Desktop/C++/clingMac/share/cling/Jupyter/kernel/clingkernel.py", line 100, in __init__
    clingInPath = shutil.which('cling')
AttributeError: 'module' object has no attribute 'which'
[W 13:44:34.272 NotebookApp] KernelRestarter: restart failed` 

I have no idea what is going on.

**Comment by dabro** [comment]
TL;DR: use Python 3.3 or higher.

Python 2.7 doesn't have `which` method on `shutil`.  See https://stackoverflow.com/questions/9877462/is-there-a-python-equivalent-to-the-which-command .

There used to be code that would catch this:
```
        except AttributeError:
            from distutils.spawn import find_executable
            whichCling = find_executable('cling')
```
and default to `find_executable`, but that seems to have been dropped in the last couple revisions.

**Comment by yiyang186** [comment]
I corrected it: clingInPath = 'path/to/cling'    ... it works, but urgly.
